cmd/minikube/cmd/delete.go

```diff
@@ -238,7 +238,7 @@ func deleteProfile(profile *pkg_config.Profile) error {
 }
 
 func deleteContext(machineName string) error {
-	if err := kubeconfig.DeleteContext(constants.KubeconfigPath, machineName); err != nil {
+	if err := kubeconfig.DeleteContext(machineName, constants.KubeconfigPath); err != nil {
```

Comment:
this is using the "Default" kubeconfig file, we should be using whatever user has passed
please use
kubeconfig.PathFromEnv() to get the Kubeconfig

Reply:
I also noticed we have many other places in the code that we are using 
"constants.KubeconfigPath"
for example
	err := kubeconfig.SetCurrentContext(profile, constants.KubeconfigPath)

we should change all of them

Reply:
This is not your fault, this was in minikube code base and this is a good opportunity to refactor all of them.

please refactor in a way that delete context does not get a PATH at all and we make sure everyone in the code base is using 
"kubeconfig.PathFromEnv()" instead of constants.KubeconfigPath

I suggest moving constants.KubeconfigPath to a Private field in "kubeconfig" package so others can not access it

Reply:
Thanks for checking that issue @medyagh 
If you don't mind, i could try to refactor to use user Path rather than "constants.KubeconfigPath" :)
Have a nice weekend !

Reply:
> please refactor in a way that delete context does not get a PATH at all and we make sure everyone in the code base is using
> "kubeconfig.PathFromEnv()" instead of constants.KubeconfigPath
> 
> I suggest moving constants.KubeconfigPath to a Private field in "kubeconfig" package so others can not access it

@medyagh 
If we move constants.KubeconfigPath to a Private field in "kubeconfig" package, then the "kubeconfig.PathFromEnv()" function also cannot access it.
Then we have to put this stirng constant in the kubeconfig package. But i think it should be placed in the constant package..
Is there any other way to prevent access from others except PathFromEnv() function ?
I have related WIP pr (#6568 )